File: code/java/model/src/main/java/modelli/Luogo.java
package modelli;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Classe modello del luogo, con tutti gli attributi nel database 
 */
public class Luogo {
	
	private String nome;
    private double latitudine;
    private double longitudine;
	private String citta;
    private String indirizzo;
    private String tipo;
    private int tempoDiVisita;
    private String immagine;
    
    @JsonCreator
	public Luogo(@JsonProperty("nome") String n, 
			@JsonProperty("latitudine") double la, 
			@JsonProperty("longitudine") double lo,
			@JsonProperty("citta") String c,
			@JsonProperty("indirizzo") String i,
			@JsonProperty("tipo") String t,
			@JsonProperty("tempoDiVisita") int tv,
			@JsonProperty("immagine") String img) {
    	this.nome = n;
    	this.latitudine = la;
    	this.longitudine = lo;
    	this.citta = c;
    	this.indirizzo = i;
    	this.tipo = t;
    	this.tempoDiVisita = tv;
    	this.immagine = img;
	}

	public String getCitta() {
		return citta;
	}

	public String getIndirizzo() {
		return indirizzo;
	}

	public String getTipo() {
		return tipo;
	}

	public String getImmagine() {
		return immagine;
	}
    
    public String getNome() {
        return nome;
    }

    public double getLatitudine() {
        return latitudine;
    }

    public double getLongitudine() {
        return longitudine;
    }

    public int getTempoDiVisita() {
        return tempoDiVisita;
    }
    
}

File: code/flutter/smart_trip_app/lib/Domain/luogo.dart
class Luogo{
  final String nome;
  final double latitudine;
  final double longitudine;
  final String city;
  final String indirizzo;
  final String tipo;
  final int tempoVisita;
  final String immagine;

   Luogo({
    required this.nome,
    required this.latitudine, 
    required this.longitudine,
    required this.city,
    required this.indirizzo,
    required this.tipo,
    required this.tempoVisita,
    required this.immagine,
  });

  //metodo di decoding
  Luogo.fromJSON(Map<String, dynamic> jsonMap) :
    nome=(jsonMap['nome'] ?? ""),
    latitudine=(jsonMap['latitudine'] ?? ""),
    longitudine=(jsonMap['longitudine'] ?? ""),
    city=(jsonMap['citta'] ?? ""),
    indirizzo=(jsonMap['indirizzo'] ?? ""),
    tipo=(jsonMap['tipo'] ?? ""),
    tempoVisita=(jsonMap['tempoDiVisita'] ?? ""),
    immagine=(jsonMap['immagine'] ?? "");


    //metodo di encoding
    Map<String, dynamic> toJson() {
    return {
      'nome': nome,
      'latitudine': latitudine,
      'longitudine': longitudine,
      'citta': city,
      'indirizzo': indirizzo,
      'tipo': tipo,
      'tempoDiVisita': tempoVisita,
      'immagine': immagine
    };
  }


      @override
String toString() {
  return '''
  Luogo(
  nome: $nome,
  latitudine: $latitudine,
  longitudine: $longitudine,
  città: $city,
  indirizzo: $indirizzo,
  tipo: $tipo,
  tempoDiVisita: $tempoVisita
)
''';
}

}